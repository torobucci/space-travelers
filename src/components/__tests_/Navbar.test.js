import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../view/Navbar';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'

jest.mock('react-bootstrap/esm/Navbar', () => 'MockedBootstrapNavbar');
describe('NavBar', () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
  });

  it('renders a logo', () => {
    const logo = screen.getByAltText('Space Travelers Hub Logo');
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('src', 'planet.png');
  });

  it('renders the brand name', () => {
    const brandName = screen.getByText('Space Travelers Hub');
    expect(brandName).toBeInTheDocument();
  });

  it('renders a link to the Rockets page', () => {
    const rocketsLink = screen.getByText('Rockets');
    expect(rocketsLink).toBeInTheDocument();
    expect(rocketsLink).toHaveAttribute('href', '/');
  });

  it('renders a link to the Missions page', () => {
    const missionsLink = screen.getByText('Missions');
    expect(missionsLink).toBeInTheDocument();
    expect(missionsLink).toHaveAttribute('href', '/missions');
  });

  it('renders a link to the My Profile page', () => {
    const profileLink = screen.getByText('My Profile');
    expect(profileLink).toBeInTheDocument();
    expect(profileLink).toHaveAttribute('href', '/my profile');
  });
  it('Navbar snapshots match', () => {
    const tree = renderer.create(<BrowserRouter>
        <NavBar />
      </BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
