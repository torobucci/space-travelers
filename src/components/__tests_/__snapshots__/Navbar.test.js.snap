// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Navbar snapshots match 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-light bg-white"
  >
    <div
      className="container"
    >
      <span
        className="d-flex align-items-center navbar-brand"
      >
        <img
          alt="Space Travelers Hub Logo"
          className="d-inline-block align-top me-3"
          height="60"
          src="planet.png"
          width="60"
        />
        <span
          className="brand-text navbar-text"
        >
          Space Travelers Hub
        </span>
      </span>
      <div
        className="ms-auto d-flex navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current="page"
          className="ms-3 active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
        <a
          className="mission ms-3"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
        <a
          className="ms-3"
          href="/my profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="divider"
  />,
]
`;
